src/Compiler/Service/FSharpParseFileResults.fs

```diff
@@ -507,6 +507,22 @@ type FSharpParseFileResults(diagnostics: FSharpDiagnostic[], input: ParsedInput,
         let result = SyntaxTraversal.Traverse(pos, input, visitor)
         result.IsSome
 
+    member _.IsPositionWithinRecordDefinition pos =
+        let isWithin left right middle =
+            Position.posGt right left && Position.posLt middle right
+
+        let visitor =
+            { new SyntaxVisitorBase<_>() with
+                override _.VisitRecordDefn(_, _, range) =
```

Comment:
Do not forget anonymous record definitions, which can exhibit exactly the same error.

Sample:
`let f (x:{|A=string|}) = x`

gives exactly the same error which this codefix targets.
(= is unexpected and should be replaced with : )

Reply:
Great point, thanks, added and tested this